Apps/Database/src/Context/GatewayDbContext.cs

```diff
@@ -387,6 +388,33 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .HasForeignKey(ad => ad.DependentHdId)
                 .OnDelete(DeleteBehavior.Cascade);
 
+            // Create Foreign key for DependentAudit to Dependent
+            modelBuilder.Entity<DependentAudit>()
+                .HasOne<Dependent>(da => da.Dependent)
+                .WithMany(d => d.DependentAudits)
+                .HasForeignKey(da => da.DelegateHdId)
+                .OnDelete(DeleteBehavior.SetNull);
```

Comment:
I'm not sure this qualifies to block the PR, however similarly with deleting the record entirely. I'm personally not a fan of losing the detail here, we would be disconnecting the action from the original target. it may feel redundant but an extra column for "TargetHdid" would ensure we don't lose the integrity of the data. And would allow easier tracing when viewing the data in an investigation post dependent deletion.

I argue, treat the table as a "warehouse table" remove the foreign key constraint (living with the performance tradeoff, potentially use an index to mitigate) and leave the data as is on deletion, or add the "duplicated" non-FK column as mentioned above.

CC: @sslaws, @ytqsl

Reply:
This is no longer an issue. Discussed with @sslaws  This will be a standalone table.